moveit_ros/planning/rdf_loader/src/synchronized_string_parameter.cpp

```diff
@@ -150,6 +152,8 @@ void SynchronizedStringParameter::stringCallback(const std_msgs::msg::String::Sh
   {
     parent_callback_(msg->data);
   }
-  content_ = msg->data;
+  queue_mutex_.lock();
+  queue_.push(msg->data);
+  queue_mutex_.unlock();
```

Comment:
@DLu here is where the callback for the message is.  The problem before is there was a race condition between the two places that were subscribing to the topic.

Reply:
@corycrean would you mind using a lock_guard here instead of the manual lock and unlock functions?

Reply:
On line 146 the shared data in content_ is accessed.  To avoid a separate race condition I would move that test to the code that pops from the queue.  The whole point of using a queue like this is to keep access to the variable content_ on a single thread.